### PR TITLE
[#51738] Enable HTTPX persistent connections.

### PR DESCRIPTION
**Pull request by ba1ash**
https://community.openproject.org/work_packages/51738

### CODE REVIEW

File: lib/open_project.rb [hunk 1]
```diff
@@ -44,4 +44,11 @@ module OpenProject
   def self.logger
     ::OpenProject::Logging::LogDelegator
   end
+
+  def self.httpx
+    HTTPX
+      .plugin(:persistent)
+      .plugin(:basic_auth)
+      .plugin(:webdav)
+  end
```
Comment (akabiru) [inline lib/open_project.rb:53]: 🟢 Nice unification 👍🏾  One thing that might change in the future is that not all plugins will be needed in all contexts- and I wonder if there's expense in eager loading them.  But as they say, _"we can cross that bridge when when we get there"_ 😄

**Review by akabiru — APPROVED** [review]
Nice one 👍🏾  Just one remark from me, otherwise LGTM

File: lib/open_project.rb (hunk 1 above)
Comment (mereghost) [reply lib/open_project.rb:53]: The plugins, in general, just load a couple of files and add a couple of hooks to the session, so shouldn't be an issue.

**Review by mereghost — APPROVED** [review]

File: lib/open_project.rb (hunk 1 above)
Comment (ba1ash) [reply lib/open_project.rb:53]: Yes, it should not be an issue and plugins code is loaded once we create a session like that for the first time.

**Pull request merged**